server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndexer.java

```diff
@@ -95,35 +100,36 @@ public void index(Iterator<IssueDoc> issues) {
     doIndex(createBulkIndexer(false), issues);
   }
 
-  private long doIndex(BulkIndexer bulk, long lastUpdatedAt, @Nullable String projectUuid) {
-    try (DbSession dbSession = dbClient.openSession(false)) {
-      IssueResultSetIterator rowIt = IssueResultSetIterator.create(dbClient, dbSession, lastUpdatedAt, projectUuid);
-      long maxDate = doIndex(bulk, rowIt);
-      rowIt.close();
-      return maxDate;
+  public void index(Collection<String> issueKeys) {
+    doIndex(createBulkIndexer(false), issueKeys);
+  }
+
+  private void doIndex(BulkIndexer bulk, Collection<String> issueKeys) {
+    try (IssueIterator issues = issueIteratorFactory.createForIssueKeys(issueKeys)) {
+      doIndex(bulk, issues);
     }
   }
 
-  private long doIndex(BulkIndexer bulk, Iterator<IssueDoc> issues) {
+  private void doIndex(BulkIndexer bulk, @Nullable String projectUuid) {
+      try (IssueIterator issues = issueIteratorFactory.createForProject(projectUuid)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)